### ISSUE TITLE
x/text/currency: add support for integer

### ISSUE THREAD

**Issue by YunxiangHuang** [golang/go#32769]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.12.6 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/yunxiang.huang/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/yunxiang.huang/Work/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/Cellar/go/1.12.6/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.12.6/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/nz/kfm5t8n50dq8gz79ln3jfchc0000gn/T/go-build553255337=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

```golang
package main

import (
	"fmt"

	"golang.org/x/text/currency"
)

func main() {
	cur := currency.MustParseISO("CNY")
	fmt.Println(currency.NarrowSymbol(cur.Amount(0)))
	fmt.Println(currency.Symbol(cur.Amount(0)))
}

```

### What did you expect to see?

```
￥0.00
CN¥ 0.00
```

### What did you see instead?

```
¥ %!f(int=00)
CN¥ %!f(int=00)
```

**Comment by andybons** [comment]
@mpvl

**Comment by rhnvrm** [comment]
@andybons @mpvl @YunxiangHuang - I'd be happy to work on this issue

**Comment by elliotwutingfeng** [comment]
This appears to be fixed as of Go 1.19
